src/main/java/lotto/domain/LottoGenerator.java

```diff
@@ -12,23 +12,43 @@ public class LottoGenerator {
     private static final int LOTTO_NUMBER_MAXIMUM = 45;
     private static final int LOTTO_TICKET_SIZE = 6;
 
-    private final List<LottoNumber> lottoNumbers;
+    private static final List<LottoNumber> lottoNumbers = IntStream.rangeClosed(LOTTO_NUMBER_MINIMUM, LOTTO_NUMBER_MAXIMUM)
+            .mapToObj(LottoNumber::new)
+            .collect(Collectors.toList());
 
-    public LottoGenerator() {
-        this.lottoNumbers = IntStream.rangeClosed(LOTTO_NUMBER_MINIMUM, LOTTO_NUMBER_MAXIMUM)
-                .mapToObj(LottoNumber::new)
+    private LottoGenerator() {
+    }
+
+    public static LottoTickets issue(int totalLottoCount, List<List<Integer>> manualNumbers) {
+        int autoLottoCount = calculateAutoLottoCount(totalLottoCount, manualNumbers);
+
+        List<List<LottoNumber>> autoLottoNumbers = generateAutoLottoNumbers(autoLottoCount);
+        List<List<LottoNumber>> manualLottoNumbers = manualNumbers.stream()
+                .map(LottoGenerator::toLottoNumbers)
+                .collect(Collectors.toList());
+
+        manualLottoNumbers.addAll(autoLottoNumbers);
+        return LottoTickets.of(manualLottoNumbers);
+    }
```

Comment:
LottoGenerator가 LottoTicket을 생성하는 Factory 역할을 하고 있네요. 수동번호도 Factory에서 생성하는게 자연스러워 보입니다. 만약, 자동로또를 위와 같은 Factory가 아닌 직접 객체를 생성하는 형태로 설계하셨다면, 수동번호의 로또 티켓을 발급하는 클래스를 구성하는게 자연스러웠을거구요.

```suggestion
    public static LottoTickets manual(List<List<Integer>> manualNumbers) {
        List<List<LottoNumber>> manualLottoNumbers = manualNumbers.stream()
                .map(LottoGenerator::toLottoNumbers)
                .collect(Collectors.toList());

        return LottoTickets.of(manualLottoNumbers);
    }

    public static LottoTickets auto(int autoLottoCount) {
        return LottoTickets.of(generateAutoLottoNumbers(autoLottoCount));
    }
```

- 자동 로또와 수동 로또 메소드를 분리하는 것은 어떨까요? 그럼, 각 메소드에 대한 테스트를 작성하기도 수월할거라 생각해요.
- `List<LottoNumber>`를 가지는 LottoNumbers 일급콜렉션을 작성하는건 어떨까요?